pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -207,10 +217,46 @@ func getParseFunc(log []byte) (parseFunc, error) {
 	return nil, fmt.Errorf("unsupported log format: %q", log)
 }
 
+// extractStreamFromLog parses a single log line and returns the stream type.
+func extractStreamFromLog(log []byte) (runtimeapi.LogStreamType, error) {
+	var stream runtimeapi.LogStreamType
+
+	if bytes.HasPrefix(log, []byte{'{'}) {
```

Comment:
Doing this check in this method is expensive (one check per line, when this should never change on a file by file perspective).  I would expect us to do this no more than once per file, at most.

What is the estimated end of life for docker log files?  Is this an option we support on the container runtime, or a consequence of dockershim support (and thus no longer needed)?

If we are still planning on supporting docker json format, then this needs to be more efficient.

Reply:
> I would expect us to do this no more than once per file, at most.

Yeah I think it makes sense, I would update this function to parse once per file.

> Is this an option we support on the container runtime, or a consequence of dockershim support 

IIUC the log files is managed by the container runtime, dockershim just translate the API calls, so I guess we need to support docker json log format.